src/_pytest/fixtures.py

```diff
@@ -593,6 +595,39 @@ def _get_active_fixturedef(
             raise FixtureLookupError(argname, self)
         fixturedef = fixturedefs[index]
 
+        # Check for attempted use of an async fixture by a sync test
+        # `self.scope` here is not the scope of the requested fixture, but the scope of
+        # the requester.
+        if (
+            self.scope == "function"
+            and not inspect.iscoroutinefunction(self._pyfuncitem.obj)
+            and (
+                inspect.iscoroutinefunction(fixturedef.func)
+                or inspect.isasyncgenfunction(fixturedef.func)
+            )
+        ):
+            if fixturedef._autouse:
+                warnings.warn(
+                    PytestRemovedIn9Warning(
+                        "Sync test requested an async fixture with autouse=True. "
```

Comment:
Add the test name to "Sync test '{name}'" and the fixture name to "async fixture '{name}'" in the phrase here to help users understand the problem better.

We also should add an entry to "deprecations", with the rationale for this and guiding users on how to update their code (installing plugins, changing the `async` fixture, etc).

Reply:
Was confused for a second what you meant with "installing plugins". The way the error for async test functions handles it is by printing a long message recommending async test plugins, maybe this message should do the same. Not sure it has much of a place in the deprecations doc - if a user has a test suite that currently works the fix almost surely whouldn't be to install a new async test plugin.